src/Compilers/Core/Portable/Symbols/IMethodSymbol.cs

```diff
@@ -214,6 +215,18 @@ public interface IMethodSymbol : ISymbol
         /// </summary>
         ImmutableArray<AttributeData> GetReturnTypeAttributes();
 
+        /// <summary>
+        /// The calling convention enum of the method symbol.
+        /// </summary>
+        SignatureCallingConvention CallingConvention { get; }
```

Comment:
>SignatureCallingConvention [](start = 8, length = 26)

Does this mean consumers need to link to System.Reflection.Metadata.dll? Did the public API rely on any types from that assembly previously?

Reply:
We depend on it, so it should come along for the ride anyway. But no, to my knowledge, we did not depend on types from SRM in the public API prior. I discussed this with Aleksey prior to him leaving, and we felt it was best to make use of the existing API that SRM uses and exactly covers the use case, rather than creating our own enum that just mirrors the SRM enum.

---
In reply to: [467315702](https://github.com/dotnet/roslyn/pull/46467#discussion_r467315702) [](ancestors = 467315702)

Reply:
Followup of offline discussion: We already expose SRM types through emit options, this is nothing new.

---
In reply to: [467316200](https://github.com/dotnet/roslyn/pull/46467#discussion_r467316200) [](ancestors = 467316200,467315702)